MovingBlocks/DestinationSol
# Waypoint placement fix when rotated

# Description
> This PR fixes #557. That is, the waypoint is placed correctly even if the map is rotated. The PR also did some refactoring to extract the "screen click position to world position"-logic and added new tests for this logic.

# Testing
> There exist 9 new tests in [ScreenToWorldMapperNoRotationTest.java](engine/src/test/java/org/destinationsol/screens/ScreenToWorldMapper/ScreenToWorldMapperNoRotationTest.java) and [ScreenToWorldMapperRotationTest.java](engine/src/test/java/org/destinationsol/screens/ScreenToWorldMapper/ScreenToWorldMapperRotationTest.java).
To test if the PR work during gameplay, you can do the following:
* Start a new game.
* Place a marker on the map and see that is placed correctly.
* Go to a planet to introduce rotation of the world.
* Place a marker again and see that it is placed correctly.

# Notes
> **OBS** The commits have a linked issues/PR. These are from the forked repo but GitHub seems to confuse them with the original repo. These linked issue/PR should be ignored.

### Pre Pull Request Checklist:

- [x] Code has been scanned with [SonarLint](http://www.sonarlint.org/intellij/)
- [x] There are no errors present in the project
- [x] Code has been formatted and indented
- [x] Methods have appropriate Javadoc ([How to write Javadoc](http://www.oracle.com/technetwork/java/javase/documentation/index-137868.html))
